### PR TITLE
feat: condition-builder支持selectMode为chained的选项层级显示

### PR DESCRIPTION
**Pull request by sqzhou**
### <samp>🤖 Generated by Copilot at 3fcb916</samp>

Added a new feature for selecting conditions in a chained mode in the `ConditionBuilder` component. This feature allows the user to choose fields from a tree structure that depends on the previous conditions. Refactored some components and interfaces to support this feature and improve code quality.

### <samp>🤖 Generated by Copilot at 3fcb916</samp>

> _Oh we're the coders of the sea, and we work on the UI_
> _We add new features and refactor, to make the users satisfied_
> _We've built a `selectMode` for `chained`, to pick fields from a tree_
> _And we track the `deepIndex` of groups, with a one, two, three_

### <samp>🤖 Generated by Copilot at 3fcb916</samp>

*  Add a new option for `selectMode` to the condition builder components, which allows the user to select a field from a hierarchical tree structure with multiple levels ([link](https://github.com/baidu/amis/pull/7120/files?diff=unified&w=0#diff-29fd0fbffb170c92ea3fd1307de3d273b553987a216926509987af10238e35d9L53-R53), [link](https://github.com/baidu/amis/pull/7120/files?diff=unified&w=0#diff-06cb9de8b8b0dfe9a40336f972f8ed3c4a47b554b0e7b88111e052cdabab00c0L33-R40), [link](https://github.com/baidu/amis/pull/7120/files?diff=unified&w=0#diff-84402e0383236ea88080c3b736a89c43538f9859b2f8594763c9b90e5d9c84dbL45-R47), [link](https://github.com/baidu/amis/pull/7120/files?diff=unified&w=0#diff-e739d54e710cb7e461bb442b0b438b9e44a2ce1f4492a64c16fe1afcd2f5d166L31-R33), [link](https://github.com/baidu/amis/pull/7120/files?diff=unified&w=0#diff-bd3cef1dd1adf5cbac8db6825eccda213b5450966cc365712f22a32093048fc6L56-R56), [link](https://github.com/baidu/amis/pull/7120/files?diff=unified&w=0#diff-504cb0e75d6206da3bb94e7bec394d9cd695d3dc89178058a5731d483f54e6b9L45-R45)).

### CODE REVIEW

File: packages/amis-ui/src/components/condition-builder/Field.tsx [hunk 1]
```diff
@@ -176,4 +177,87 @@ export class ConditionField extends React.Component<
...
+    values.splice(index, values.length - index);
+    value && values.push(value);
+    const stacks = this.computedStask(values);
+    this.setState(
+      {
+        stacks,
+        values
+      },
+      () => {
+        this.props?.onChange(value);
```
Comment (lurunze1226) [inline packages/amis-ui/src/components/condition-builder/Field.tsx:211]: props是肯定存在的

File: packages/amis-ui/src/components/condition-builder/Field.tsx [hunk 2]
```diff
@@ -12,38 +13,38 @@ import {
...
 import SearchBox from '../SearchBox';
 import TreeSelection from '../TreeSelection';
+import {Options} from '../Select';
 import {SpinnerExtraProps} from '../Spinner';
 
-export interface ConditionFieldProps
-  extends ThemeProps,
-    LocaleProps,
-    SpinnerExtraProps {
+const getFlatOptions = (options: Options) => {
```
Comment (lurunze1226) [inline packages/amis-ui/src/components/condition-builder/Field.tsx:25]: 这个直接挂在ConditionField Class里吧，其他地方似乎也没用到

**Review by lurunze1226 — CHANGES_REQUESTED** [review]

File: docs/zh-CN/components/form/condition-builder.md [hunk 3]
```diff
@@ -500,7 +500,7 @@ type Value = ValueGroup;
 
 > 2.3.0 及以上版本
 
-通过 selectMode 配置组合条件左侧选项类型，可配置项为`list`、`tree`，默认为`list`。两者数据格式相同，只是下拉框展示方式不同，当存在多层 children 嵌套时，建议使用`tree`。
+通过 selectMode 配置组合条件左侧选项类型，可配置项为`list`、`tree`、`chained`，默认为`list`。这三个数据格式基本类似，只是下拉框展示方式不同，`tree`是树形下拉，`chained`为多个级连的下拉。当存在多层 children 嵌套时，建议使用`tree`。
```
Comment (lurunze1226) [inline docs/zh-CN/components/form/condition-builder.md]: 级联

**Review by lurunze1226 — APPROVED** [review]

File: packages/amis-ui/src/components/ChainedDropdownSelection.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+> {
+  constructor(props: ChainedDropDownSelectionProps) {
+    super(props);
+
+    this.state = this.computed(props.value, props.options);
+  }
+
+  componentDidUpdate(prevProps: ChainedDropDownSelectionProps) {
+    const {options, value} = this.props;
+    if (options !== prevProps.options) {
```
Comment (2betop) [inline packages/amis-ui/src/components/ChainedDropdownSelection.tsx:46]: 或者 value 变化都要重新同步到 state

File: packages/amis-ui/src/components/DropDownSelection.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    this.setState({searchText: txt});
+  }
+
+  filterOptions(options: any[]) {
+    const {valueField = 'value'} = this.props;
+    const txt = this.state.searchText;
+    if (!txt) {
+      return this.props.options;
+    }
+    return options
```
Comment (2betop) [inline packages/amis-ui/src/components/DropDownSelection.tsx:64]: 建议参考 transfer 中实现，改成 filterTree + matchSorter https://github.com/baidu/amis/blob/master/packages/amis/src/renderers/Form/Transfer.tsx#L390-L402

File: packages/amis-ui/src/components/condition-builder/Group.tsx [hunk 6]
```diff
@@ -196,6 +203,23 @@ export class ConditionGroup extends React.Component<
           : value!.children
         : null;
 
+    let addConditionVisibleBool = true;
+    let addConditionGroupVisibleBool = true;
+
+    const param = {depth, breadth: body?.length ?? 0};
```
Comment (2betop) [inline packages/amis-ui/src/components/condition-builder/Group.tsx:209]: 这个功能应该在 renderer 那层实现，component 这层只加钩子，比如 isConditionVisible?(context)，renderer 那层再去支持表达式运算，同时把 props.data 的上下文也扩展进去

**Review by 2betop — APPROVED** [review]

**Pull request merged**